File: accommodation_service/src/main/java/huy/project/accommodation_service/core/usecase/GetImageUseCase.java
package huy.project.accommodation_service.core.usecase;

import huy.project.accommodation_service.core.domain.entity.ImageEntity;
import huy.project.accommodation_service.core.port.IImagePort;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@RequiredArgsConstructor
public class GetImageUseCase {
    private final IImagePort imagePort;;

    public List<ImageEntity> getImagesByEntityIdAndType(Long entityId, String entityType) {
        return imagePort.getImagesByEntityIdAndType(entityId, entityType);
    }
}

File: location_service/LocationService.Core/UseCase/GetImageUseCase.cs
using LocationService.Core.Domain.Entity;
using LocationService.Core.Port;

namespace LocationService.Core.UseCase
{
    public interface IGetImageUseCase
    {
        Task<List<ImageEntity>> GetImagesByEntityIdAndEntityType(long entityId, string entityType);
    }

    public class GetImageUseCase : IGetImageUseCase
    {
        private readonly IImagePort _imagePort;

        public GetImageUseCase(IImagePort imagePort)
        {
            _imagePort = imagePort;
        }

        public async Task<List<ImageEntity>> GetImagesByEntityIdAndEntityType(long entityId, string entityType)
        {
            return await _imagePort.GetImagesByEntityIdAndEntityType(entityId, entityType);
        }
    }
}